scripts/ai2-internal/run_with_beaker.py

```diff
@@ -12,9 +12,7 @@
 from allennlp.commands.train import Train
 from allennlp.common.params import Params
 
-#TODO(michaels): add CLI support for mounting datasets.
-
-def main(param_file, description):
+def main(param_file, extra_beaker_commands):
     ecr_repository = "896129387501.dkr.ecr.us-west-2.amazonaws.com"
     commit = subprocess.check_output(["git", "rev-parse", "HEAD"], universal_newlines=True).strip()
     image = f"{ecr_repository}/allennlp/allennlp:{commit}"
```

Comment:
I missed this in the previous PR for this script, but I think you need to add a random hash as in the previous script onto the end of the git commit here, otherwise when the image gets pulled from ECR it could be different if someone else has also pushed an image with the same name (assuming beaker pulls the `latest` image by default). Or maybe that doesn't matter now, as beaker pulls the image almost immediately after you push it. Still, seems safer to add the hash unless there's something i'm missing

Reply:
If it's easy, it'd be even better if you can say whether the working tree is clean or dirty, too.  If it's not easy, it's probably not worth the effort, though.

Reply:
@DeNeutoy I agree we should add a hash.  The old script's random number [was broken](https://github.com/allenai/allennlp/pull/757/files#diff-91c872fd064a4d494d631616c1eee34fL17) so I skipped this in my last PR (I wondered if I were missing something).

It's easy to tell whether the working tree is clean or dirty.  This is [how I handled it](https://github.com/allenai/allennlp/pull/83/files#diff-8dd2caa4ff5cd68e3b52d78173f5cc7aR28) in a PR from way back when.  However, I (and I suspect our users) often run Beaker experiments in dirty git trees.  I worry this feature would be annoying, and I want to make sure this command isn't annoying so people actually use it.

I'm uncertain about a few parts of this script.  I'm considering splitting it into two--one which builds images and one which runs images.

Reply:
Oh, yes, I definitely run things with dirty trees sometimes.  I wasn't suggesting crashing if it's dirty, just including either "dirty" or "clean" in the image tag, before the random number, as an extra piece of info if you're looking at a beaker experiment.

Reply:
@matt-gardner good suggestion.  I'm now thinking of appending a random number iff it's dirty.

@DeNeutoy `RANDOM` was undefined.  The image names it created ended with a hyphen.

Reply:
Oh interesting - on my machine $RANDOM generates a random number every time. :man_shrugging: